HCollections/HListT.fs

```diff
@@ -6,11 +6,11 @@ open TypeEquality
 [<NoEquality>]
 type HListT<'ts, 'elem> =
     private
-    | Empty of Teq<'ts, unit>
-    | Cons of HListTConsCrate<'ts, 'elem> * 'ts TypeList
+    | Empty of 'ts HList * Teq<'ts, unit>
+    | Cons of 'ts HList * 'elem list * HListTConsCrate<'ts, 'elem>
 
 and private HListTConsEvaluator<'ts, 'elem, 'ret> =
-    abstract member Eval<'t, 'ts2> : 't -> 'elem -> HListT<'ts2, 'elem> -> Teq<'ts, 't -> 'ts2> -> 'ret
```

Comment:
Don't you want access to the things you consed on here? I think the original HListTConsEvaluator API is clearer

Reply:
Why would we want to store them? We have them on the actual lists we now store